File: Models/UserModel.cs
using Newtonsoft.Json;

namespace ManabieTodo.Models
{
    public class UserModel
    {
        [JsonProperty("ID")]
        public int Id { get; set; } = 0;
        [JsonProperty("NAME")]
        public string? Name { get; set; }
        [JsonProperty("USERNAME")]
        public string? Username { get; set; }
        [JsonProperty("PASSWORD")]
        public string? Password { get; set; } = null;
        [JsonProperty("ALLOWED_TASK_DAY")]
        public int AllowedTaskDay { get; set; } = (new Random()).Next(1, 10);
    }

    public class LoginModel
    {

    }
}

File: models/user_model.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"golang.org/x/crypto/bcrypt"
	//
)

type User struct {
	Id       primitive.ObjectID `json:"id,omitempty"`
	Username string             `json:"username,omitempty" validate:"required,min=4,max=25"`
	Password string             `json:"password,omitempty" validate:"required"`
	Name     string             `json:"name,omitempty" validate:"max=16"`
	Limit    int                `json:"limit"`
	Status   bool
	Role     string
}

func HashPassword(password string) (string, error) {
	bytes, err := bcrypt.GenerateFromPassword([]byte(password), 14)
	return string(bytes), err
}

func CheckPasswordHash(password, hash string) bool {
	err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
	return err == nil //true or false
}
